compiler/rustc_middle/src/dep_graph/dep_node.rs

```diff
@@ -271,36 +175,14 @@ impl DepNodeExt for DepNode {
     /// refers to something from the previous compilation session that
     /// has been removed.
     fn extract_def_id<'tcx>(&self, tcx: TyCtxt<'tcx>) -> Option<DefId> {
-        if self.kind.fingerprint_style(tcx) == FingerprintStyle::DefPathHash {
+        if tcx.fingerprint_style(self.kind) == FingerprintStyle::DefPathHash {
             Some(tcx.def_path_hash_to_def_id(DefPathHash(self.hash.into()), &mut || {
```

Comment:
This can't be moved to `rustc_query_system` without adding `def_path_hash_to_def_id` to the `DepContext` trait; I can do that, but not sure it was worth the churn. Let me know and I'm happy to change it.